models/project.py

```diff
@@ -0,0 +1,64 @@
+from google.appengine.ext import ndb
+
+
+class Project(ndb.Model):
+
+    title = ndb.StringProperty()
+    member_keys = ndb.KeyProperty(repeated=True, kind='User')
+    task_keys = ndb.KeyProperty(repeated=True, kind='Task')
+    course = ndb.KeyProperty()
+
+    @property
+    def members(self):
+        return map(lambda k: k.get(), self.member_keys)
+
+    def tasks(self):
```

Comment:
We need to add `@property` here, so that it looks like this:

```python
@property
def tasks(self):
    return map(lambda k: k.get(), self.task_keys)
```

It's probably not obvious what this is doing, between the combination of `@property` and the `lambda` function, so I'll explain just in case;  ideally, we want to be able to access `project.tasks` and have that result in an array of task objects.  However, NDB doesn't allow us to do this; we get an array of keys instead if we access `project.task_keys`.  Doing the `key -> object` translation over and over is really annoying, so we want to define a "getter" property that will run some code for us, allowing to create the interface that we want.

`@property` is used to define a "computed property" on an object, which is sort of halfway between a method and a property.  While you access it like a property, it can do some calculations and returns the result, instead of just being a variable value.  In this case, we use `map` to iterate over each key in `task_keys` and then use a lambda function to call `.get()` on each of them, which is how NDB is able to look up some object by a key.  Thus, we create a "computed property" that fetches an array of real tasks instead of just the keys.  However, without `@property`, Python doesn't know that that's what we're trying to do and instead creates a method called `.tasks()`.